src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -318,4 +318,13 @@ object messages {
         |$code2
         |""".stripMargin
   }
+
+  case class IllegalExpressionStart()(implicit ctx: Context) extends Message(14) {
+    val kind = "Syntax"
+    val msg = "illegal start of simple expression"
+    val explanation = {
+      hl"""|The expression you've provided is not valid because
+           |an expression in Scala always returns a value of some type""".stripMargin
```

Comment:
It won't produce an exception if the token is defined no - so it shouldn't be a problem. Good thinking on including user code :+1: